### DISCUSSION TITLE
Is Isar now a SQLite wrapper or NoSQL solution?

### DISCUSSION THREAD

**Discussion by ebelevics** [isar/isar#1414] [Help]
So far SQLite seems to be the most robust solution yet (especially when considering data aggregation) for database solutions. While Sqlfite is decent enough, managing JOINS, relationships and Raw text queries is usually a tedious process, especially when you have complex and deep data models. 
I was also considering making my own wrapper around sqlite3_flutter_libs, did worked for 2 days, but then got stuck that code generation is a must for ease of use. I was considering on that, but then I noticed something interesting in Isar solution.

I came back to check how is Isar doing and noticed Isar SQLite in benchmarks, and got confused. Is Isar now a NoSQL or SQL solution, or both? Can I consider it now as SQL ORM, or am I missing something?

If that so, it makes me really excited, as most problems/solutions surrounding SQLite are already present. But also that is a quite of shift and most likely will not come any time soon, not to mention bug fixes during production phase.

P.S. I have played with Isar v4 and to be honest, I am surprisingly satisfied. I can get large list of data and do calculations or groupBy with large List objects. But didn't find any other relationship possibility except @embedded. How do you join together two @collections, or is not yet implemented?

P.P.S.
Took a deeper look in library and generator logic, seems like only relationship available for now is Embedded, also `isar_core/src/sqlite/` didn't have any join statements
```dart
  /// list of embedded objects (6 + length * (6 + size) bytes)
  objectList('ObjectList');
```

**Comment by simc** [comment]
Isar has two storage engines. The ultra fast Isar engine and the SQLite engine which is basically a wrapper around SQLite.

You can switch between the two with single line of code.

**Reply by tumble29** [reply]
How to switch from one engine to another? I couldn't find this in the documentation.

**Reply by itsJoKr** [reply]
@tumble29 

in v4
```dart
Isar.open(
        engine: IsarEngine.sqlite,

or

Isar.open(
        engine: IsarEngine.isar,
```

**Reply by KiddoV** [reply]
V4 is still not in production. Sadly :(
